### PR TITLE
sci-electronics/iverilog: add iverilog-9999.ebuild

### PR DESCRIPTION
**Pull request by vowstar**
* Found a better way to fix the build race condition, fixed in this ebuild
* Removed ``` .NOTPARALLEL: install``` to speed up build, and workaround parallel build bug
* Applied these fix in iverilog-10.3.ebuild

Upstream fix by @minux: https://github.com/steveicarus/iverilog/pull/294
Issue related: https://github.com/gentoo/gentoo/pull/14096

Package-Manager: Portage-2.3.84, Repoman-2.3.20
Signed-off-by: Huang Rui <vowstar@gmail.com>

### CODE REVIEW

File: sci-electronics/iverilog/iverilog-10.3.ebuild [hunk 1]
```diff
@@ -57,8 +54,20 @@ src_prepare() {
...
+
 src_install() {
 	local DOCS=( *.txt )
+	# Fix build fail problem when using large job number, make it parallel safe
+	# See also https://github.com/gentoo/gentoo/pull/14096
+	# and https://github.com/vowstar/vowstar-overlay/issues/1
+	# NOTE: this is only a workaround for upstream parallel build bug
+	# this fix made by minux https://github.com/minux
+	# https://github.com/steveicarus/iverilog/pull/294
+	# Due to this issue, we must run ``emake installdirs`` before ``default``
```
Comment (juippis) [inline sci-electronics/iverilog/iverilog-10.3.ebuild]: Too much text here, the first line and upstream issue link is enough. Ie, 
```
# Default build fails with parallel jobs, 
# https://github.com/steveicarus/iverilog/pull/294
```
That's enough information.

File: sci-electronics/iverilog/iverilog-10.3.ebuild [hunk 2]
```diff
@@ -57,8 +54,20 @@ src_prepare() {
...
+	emake installdirs DESTDIR="${D}" || die 'emake failed'
```
Comment (juippis) [inline sci-electronics/iverilog/iverilog-10.3.ebuild]: `emake` has a built-in `die` function, so you're duplicating code here. These comments apply to -9999 ebuild as well.

Rest looks good and glad you found a solution to parallel install so fast!

File: sci-electronics/iverilog/iverilog-10.3.ebuild (hunk 1 above)
Comment (vowstar) [reply sci-electronics/iverilog/iverilog-10.3.ebuild]: Thank you! I've applied it to both 10.3 and 9999 ebuild

File: sci-electronics/iverilog/iverilog-10.3.ebuild (hunk 2 above)
Comment (vowstar) [reply sci-electronics/iverilog/iverilog-10.3.ebuild]: > `emake` has a built-in `die` function, so you're duplicating code here. These comments apply to -9999 ebuild as well.
> 
> Rest looks good and glad you found a solution to parallel install so fast!

Thank you! I've remove `` || die 'emake failed'`` in both 10.3 and 9999 ebuild

**Comment by juippis** [comment]
I split your commit into two, please test that 10.3 still works. I removed `src_compile() { default }` because it should be the default behaviour without defining it.

**Comment by vowstar** [comment]
> I split your commit into two, please test that 10.3 still works. I removed `src_compile() { default }` because it should be the default behaviour without defining it.

Thank you very much!

I tested, after removed ``src_compile() { default }``, both 10.3 and 9999 are works. 
The ``install`` target in Makefile run the ``all`` target correctly.

**Pull request closed without merging**